formfactory/views.py

```diff
@@ -31,6 +33,11 @@ def get_form(self, form_class=None):
         return self.form_object.as_form()
 
     def get_success_url(self):
+        redirect_url = self.request.GET.get(
+            SETTINGS["redirect-url-param-name"]
+        )
+        if redirect_url:
+            return redirect_url
         return self.request.META.get(
             "HTTP_REFERER", self.request.META.get("PATH_INFO")
```

Comment:
Rather use request.path_info. It does some work behind the scenes and delivers better results than hitting META directly.